subsystems/networking/networkmanager.go

```diff
@@ -720,6 +727,10 @@ func (w *Provisioning) mainLoop(ctx context.Context) {
 					pModeChange = w.connState.getProvisioningChange()
 				}
 			}
+
+			if shouldReboot && w.doReboot(ctx) {
```

Comment:
This and below are a few changes for the reboot functionality. Everything else in this file is just renames of fields/structs/etc.